src/clr-addons/clr-control-warning/if-warning.directive.ts

```diff
@@ -0,0 +1,106 @@
+import { AfterViewInit, Directive, ElementRef, Input, Renderer2, TemplateRef, ViewContainerRef } from '@angular/core';
+
+@Directive({
+  selector: '[clrIfWarning]',
+})
+export class ClrIfWarning implements AfterViewInit {
+  private hostElement: Element;
+  private inputElement: Element;
+  private inputGroup: Element;
+  private comboboxWrapper: Element;
+  private focusIndicator: Element;
+  private textArea: Element;
+  private icon = this.renderer.createElement('cds-icon');
+
+  constructor(
+    private container: ViewContainerRef,
+    private template: TemplateRef<any>,
+    private renderer: Renderer2,
+    private host: ElementRef
+  ) {}
+
+  ngAfterViewInit(): void {
+    this.icon.setAttribute('shape', 'exclamation-triangle');
+    this.renderer.addClass(this.icon, 'clr-control-warning-icon');
+  }
+
+  @Input() set clrIfWarning(clrIfWarning: boolean) {
+    if (clrIfWarning) {
+      this.container.createEmbeddedView(this.template);
+      setTimeout(() => {
+        this.hostElement = this.host.nativeElement?.previousElementSibling;
+        const wrapper = this.hostElement?.previousElementSibling;
+        this.setComponents();
+        //Radio buttons have no wrapper
+        if (!wrapper) {
+          this.renderer.insertBefore(this.hostElement?.parentElement, this.icon, this.hostElement);
+        }
+        //if an error icon is already present, place the icon in the parent
+        if (wrapper?.classList?.contains('clr-validate-icon')) {
+          this.renderer.insertBefore(wrapper.parentElement, this.icon, wrapper?.nextSibling);
+        } else {
+          this.renderer.insertBefore(wrapper, this.icon, wrapper?.previousElementSibling);
+        }
+        this.setControlStyles();
+      });
+    } else {
+      this.resetControlStyles();
+      this.icon.remove();
+      this.container.clear();
+    }
+  }
+
+  setComponents() {
```

Comment:
puh, thats a lot of special case handling tbh.
don't we try to add it into clarity itself to remove all that glue code which just exists, because it should belong to clarity and not addons? :)

because it needs some internal knowledge of how clarity creates its components to work correctly.